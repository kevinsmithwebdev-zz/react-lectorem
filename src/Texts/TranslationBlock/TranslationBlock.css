.TranslationBlock {
  font-style: italic;
  font-size: 22px;
  flex: 1;
  text-align: left;
}

.TranslationBlock.below {
  padding: 5px 0 0 0;
}

.TranslationBlock.beside {
  padding: 0 0 0 20px;
}
