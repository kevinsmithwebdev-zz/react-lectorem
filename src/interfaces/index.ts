export interface LectInterface {
  start?: number,
  end?: number,
  text: string,
  translation?: string,
  explanation?: string,
}

export interface SentenceInterface {
  lects: LectInterface[],
  sentenceTranslation?: string,
}

export interface HeaderInterface {
  imagePath?: string,
}

export interface ParagraphInterface {
  header?: HeaderInterface,
  sentences: SentenceInterface[],
}

export interface StoryDataInterface {
  audioPath: string,
  title?: string,
  subtitle?: string,
  paragraphs: ParagraphInterface[],
}

export interface ConfigurationInterface {
  shouldShowTranslation?: boolean,
  isTranslationBelow?: boolean,
  isBySentence?: boolean,
}