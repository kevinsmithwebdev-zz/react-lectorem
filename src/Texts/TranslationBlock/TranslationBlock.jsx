import React from 'react';
import './TranslationBlock.css';

const aggregateSentences = sentences => sentences.map(s => s.sentenceTranslation).join('  ');

export const getClassName = isTranslationBelow => ['TranslationBlock', isTranslationBelow ? 'below' : 'beside'].join(' ');

const TranslationBlock = (
  { block: { sentences }, configuration: { shouldShowTranslation, isTranslationBelow } },
) => {
  if (!shouldShowTranslation) {
    return null;
  }

  const className = getClassName(isTranslationBelow);
  return (
    <div className={ className }>
      { aggregateSentences(sentences) }
    </div>
  );
};

export default TranslationBlock;
