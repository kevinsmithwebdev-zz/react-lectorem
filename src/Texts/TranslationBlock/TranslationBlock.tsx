import React from 'react';
import classes from './TranslationBlock.module.css';
import { SentenceInterface, ConfigurationInterface } from '../../interfaces/index';

interface TranslationBlockInterface {
  block: {
    sentences: SentenceInterface[],
  },
  configuration: ConfigurationInterface,
}

export const aggregateTranslations = (sentences: SentenceInterface[]): string => sentences.map(s => s.sentenceTranslation).join('  ');

export const getClassName = (isTranslationBelow: boolean): string => [classes.TranslationBlock, isTranslationBelow ? classes.below : classes.beside].join(' ');

const TranslationBlock: React.FC<TranslationBlockInterface> = (
  { block: { sentences }, configuration: { isTranslationShown, isTranslationBelow } }
) => {
  if (!isTranslationShown) {
    return null;
  }

  const className = getClassName(isTranslationBelow);
  return (
    <div className={ className }>
      { aggregateTranslations(sentences) }
    </div>
  );
};

export default TranslationBlock;
