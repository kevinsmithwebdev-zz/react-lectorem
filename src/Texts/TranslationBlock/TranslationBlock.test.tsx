import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TranslationBlock, { aggregateTranslations, getClassName } from './TranslationBlock';
import storyData from '../../fixtureData/story.data';
import { SentenceInterface } from '../../interfaces/index';

describe('TranslationBlock', () => {
  describe('the main component render', () => {
    const { sentences } = storyData.paragraphs[0];
    const block = { sentences };
    const idx = 127;
    const readTime = 3.1459;
    const showModal = () => {};
    const baseProps = { block, configuration: {}, idx, readTime, showModal };

    const renderer = ShallowRenderer.createRenderer();
    describe('with data but !isTranslationShown', () => {
      const configuration = {
        isTranslationShown: false,
        isTranslationBelow: false,
        isBySentence: false,
      };
      const props = {
        ...baseProps,
        configuration,
      };

      it('should render and match snapshot', () => {
        renderer.render(<TranslationBlock {...props} />);
        expect(renderer.getRenderOutput()).toBeNull();
      });
    });

    describe('with data and isTranslationShown', () => {
      const configuration = {
        isTranslationShown: true,
        isTranslationBelow: false,
        isBySentence: false,
      };
      const props = {
        ...baseProps,
        configuration,
      };

      it('should render and match snapshot', () => {
        renderer.render(<TranslationBlock {...props} />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      });
    });
  });

  describe('aggregateTranslations', () => {
    const sentences = [
      { sentenceTranslation: 'This is sentence 1' },
      { sentenceTranslation: 'And this is the second one.' },
      { sentenceTranslation: 'Don\'t forget the last.' },
    ] as SentenceInterface[];
    const expectedReturn = 'This is sentence 1  And this is the second one.  Don\'t forget the last.';
    const actualReturn = aggregateTranslations(sentences);
    it('should return the correct aggregated string', () => {
      expect(actualReturn).toBe(expectedReturn);
    });
  });

  describe('getClassName', () => {
    describe('when isTranslationBelow', () => {
      const isTranslationBelow = true;
      const expectedReturn = 'TranslationBlock below';
      const actualReturn = getClassName(isTranslationBelow);
      it('should return the correct string', () => {
        expect(actualReturn).toBe(expectedReturn);
      });
    });

    describe('when !isTranslationBelow', () => {
      const isTranslationBelow = false;
      const expectedReturn = 'TranslationBlock beside';
      const actualReturn = getClassName(isTranslationBelow);
      it('should return the correct string', () => {
        expect(actualReturn).toBe(expectedReturn);
      });
    });
  });
});
