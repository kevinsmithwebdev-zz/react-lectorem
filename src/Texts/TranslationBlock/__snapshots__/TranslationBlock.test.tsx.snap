// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslationBlock the main component render with data and shouldShowTranslation should render and match snapshot 1`] = `
<div
  className="TranslationBlock beside"
>
  Brother James.  Brother James.
</div>
`;
